core/cli/src/informant.rs

```diff
@@ -53,79 +53,76 @@ where C: Components {
 
 	let display_notifications = network.status().for_each(move |sync_status| {
 
-		if let Ok(info) = client.info() {
-			let best_number = info.chain.best_number.saturated_into::<u64>();
-			let best_hash = info.chain.best_hash;
-			let num_peers = sync_status.num_peers;
-			let speed = move || speed(best_number, last_number, last_update);
-			last_update = time::Instant::now();
-			let (status, target) = match (sync_status.sync.state, sync_status.sync.best_seen_block) {
-				(SyncState::Idle, _) => ("Idle".into(), "".into()),
-				(SyncState::Downloading, None) => (format!("Syncing{}", speed()), "".into()),
-				(SyncState::Downloading, Some(n)) => (format!("Syncing{}", speed()), format!(", target=#{}", n)),
-			};
-			last_number = Some(best_number);
-			let txpool_status = txpool.status();
-			let finalized_number: u64 = info.chain.finalized_number.saturated_into::<u64>();
-			let bandwidth_download = network.average_download_per_sec();
-			let bandwidth_upload = network.average_upload_per_sec();
-			info!(
-				target: "substrate",
-				"{}{} ({} peers), best: #{} ({}), finalized #{} ({}), ⬇ {} ⬆ {}",
-				Colour::White.bold().paint(&status),
-				target,
-				Colour::White.bold().paint(format!("{}", sync_status.num_peers)),
-				Colour::White.paint(format!("{}", best_number)),
-				best_hash,
-				Colour::White.paint(format!("{}", finalized_number)),
-				info.chain.finalized_hash,
-				TransferRateFormat(bandwidth_download),
-				TransferRateFormat(bandwidth_upload),
-			);
-
-			#[allow(deprecated)]
-			let backend = (*client).backend();
-			let used_state_cache_size = match backend.used_state_cache_size(){
-				Some(size) => size,
-				None => 0,
-			};
-
-			// get cpu usage and memory usage of this process
-			let (cpu_usage, memory) = if sys.refresh_process(self_pid) {
-				let proc = sys.get_process(self_pid).expect("Above refresh_process succeeds, this should be Some(), qed");
-				(proc.cpu_usage(), proc.memory())
-			} else { (0.0, 0) };
-
-			let network_state = network.network_state();
-
-			telemetry!(
-				SUBSTRATE_INFO;
-				"system.interval";
-				"network_state" => network_state,
-				"status" => format!("{}{}", status, target),
-				"peers" => num_peers,
-				"height" => best_number,
-				"best" => ?best_hash,
-				"txcount" => txpool_status.ready,
-				"cpu" => cpu_usage,
-				"memory" => memory,
-				"finalized_height" => finalized_number,
-				"finalized_hash" => ?info.chain.finalized_hash,
-				"bandwidth_download" => bandwidth_download,
-				"bandwidth_upload" => bandwidth_upload,
-				"used_state_cache_size" => used_state_cache_size,
-			);
-		} else {
-			warn!("Error getting best block information");
-		}
+		let info = client.info();
+		let best_number = info.chain.best_number.saturated_into::<u64>();
```

Comment:
While you are at it. Could you make it correct and require `BlockNumber` to implement `Serialize`?

https://github.com/paritytech/substrate/issues/2652

Reply:
That's not a trivial change. We do some rounding operations by multiplying by 10000 and dividing by a number of seconds. I'm not sure how to handle that, and suggest to do this later.